src/main.rs

```diff
@@ -45,6 +46,11 @@ fn main() -> anyhow::Result<()> {
     path.push(format!("{APP_NAME}/{APP_NAME}.db"));
     path
   };
+
+  if fs::metadata(&db_path).is_err() {
+    fs::create_dir_all(db_path.parent().unwrap())?;
+  }
```

Comment:
Also, I think it may be better to move this piece of logic inside `IndexedLocalDir::new` so we don't clutter `main` function too much.